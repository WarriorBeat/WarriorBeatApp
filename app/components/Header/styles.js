/**
 * Header/styles.js
 * Header component styles
 * components
 */

import { StyleSheet } from "react-native"
import { colors } from "config/styles"
import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from "react-native-responsive-screen"

const styles = StyleSheet.create({
  header_container: {
    flexDirection: "column",
    height: hp("30%"),
    paddingTop: hp("1.5%"),
  },
  header_image: {
    height: wp("30%"),
    width: wp("30%"),
  },
  center_container: {
    alignSelf: "stretch",
    width: "100%",
    height: "100%",
  },
  tab_container: {
    flexDirection: "row",
    position: "absolute",
    bottom: 0,
  },
  tab_button: {
    borderRadius: 0,
    height: hp("6%"),
    width: wp("50%"),
    paddingBottom: 0,
    borderBottomWidth: 4,
    borderColor: colors.transparent,
  },
  tab_button_selected: {
    borderColor: colors.secondary,
  },
  backButtonContainer: {
    position: "absolute",
    top: hp("2.5%"),
    left: 0,
    width: wp("100%"),
    height: hp("10%"),
    paddingHorizontal: wp("1%"),
    zIndex: 99,
    alignItems: "flex-start",
    justifyContent: "center",
  },
  backButton: {
    padding: wp("3%"),
    color: colors.white,
  },
})

export const config = {
  backButton: {
    underlayColor: colors.transparent,
  },
}

export default styles
