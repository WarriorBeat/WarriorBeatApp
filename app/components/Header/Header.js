/**
 * Header.js
 * Header Component
 * components
 */

import React from "react"
import PropTypes from "prop-types"
import { Header as RNEHeader, Icon } from "react-native-elements"
import { observer, inject, PropTypes as MobxTypes } from "mobx-react/native"
import { icons, colors } from "config/styles"

const Return = props => <Icon {...icons.arrow_back} underlayColor={colors.transparent} {...props} />

const Header = inject("uiStore")(
  observer(({
    uiStore, backgroundColor, activeIconColor, inActiveIconColor, active,
  }, props) => (
    <RNEHeader
      barStyle={active ? "light-content" : "dark-content"}
      backgroundColor={backgroundColor}
      leftComponent={(
        <Return
          color={active ? activeIconColor : inActiveIconColor}
          onPress={() => uiStore.goBack()}
        />
      )}
      {...props}
    />
  )),
)

Header.wrappedComponent.propTypes = {
  uiStore: MobxTypes.observableObject.isRequired,
}

Header.propTypes = {
  activeColor: PropTypes.string,
  inActiveColor: PropTypes.string,
  activeIconColor: PropTypes.string,
  inActiveIconColor: PropTypes.string,
  active: PropTypes.bool,
}

Header.defaultProps = {
  activeColor: colors.primary,
  inActiveColor: colors.transparent,
  activeIconColor: colors.white,
  inActiveIconColor: colors.primaryDark,
  active: false,
}

export default Header
