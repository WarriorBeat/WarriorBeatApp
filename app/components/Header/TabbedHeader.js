/**
 * TabbedHeader.js
 * Header Component
 * components
 */

import React from "react"
import { View, ViewPropTypes, TouchableOpacity } from "react-native"
import { Header, Button, Icon } from "react-native-elements"
import Text from "components/Text"
import { PropTypes } from "prop-types"
import { observer, inject, PropTypes as MobxTypes } from "mobx-react/native"
import { observable } from "mobx"
import Image from "react-native-fast-image"
import { colors, icons } from "config/styles"
import brandMedia from "config/assets"
import styles, { config } from "./styles"

const TabButton = ({
  title, active, onPress, pos,
}) => (
  <Button
    onPress={() => onPress(pos)}
    disabled={active === pos}
    type="clear"
    buttonStyle={styles.tab_button}
    disabledStyle={styles.tab_button_selected}
    title={(
      <Text Color="white" Weight="bold">
        {title}
      </Text>
    )}
  />
)

const Tabs = (props) => {
  const {
    leftButton, rightButton, active, onPress,
  } = props
  return (
    <View style={styles.tab_container}>
      <TabButton onPress={onPress} title={leftButton} active={active} pos="left" />
      <TabButton onPress={onPress} title={rightButton} active={active} pos="right" />
    </View>
  )
}

@inject("uiStore")
@observer
class TabbedHeader extends React.Component {
  @observable
  activeTab = "left"

  _renderCenterComponent = () => {
    const { centerImage } = this.props
    return (
      <View>
        <Image
          style={styles.header_image}
          source={centerImage}
          resizeMode={Image.resizeMode.contain}
        />
      </View>
    )
  }

  render() {
    const {
      backgroundColor, leftButton, rightButton, onTabSwitch, uiStore,
    } = this.props
    return (
      <View>
        <Icon
          {...icons.arrow_back}
          {...config.backButton}
          iconStyle={styles.backButton}
          containerStyle={styles.backButtonContainer}
          onPress={() => uiStore.goBack()}
          component={TouchableOpacity}
        />
        <Header
          barStyle="light-content"
          centerContainerStyle={styles.center_container}
          containerStyle={styles.header_container}
          backgroundColor={backgroundColor}
          centerComponent={this._renderCenterComponent()}
        />
        <Tabs
          leftButton={leftButton}
          rightButton={rightButton}
          active={this.activeTab}
          onPress={(pos) => {
            this.activeTab = pos
            return onTabSwitch(pos === "left" ? 0 : 1)
          }}
        />
      </View>
    )
  }
}

TabbedHeader.wrappedComponent.propTypes = {
  uiStore: MobxTypes.observableObject.isRequired,
}

TabbedHeader.propTypes = {
  backgroundColor: ViewPropTypes.style,
  centerImage: PropTypes.number,
  leftButton: PropTypes.string.isRequired,
  rightButton: PropTypes.string.isRequired,
  onTabSwitch: PropTypes.func.isRequired,
}

TabbedHeader.defaultProps = {
  backgroundColor: colors.primary,
  centerImage: brandMedia.warrior_head,
}

Tabs.propTypes = {
  leftButton: PropTypes.string.isRequired,
  rightButton: PropTypes.string.isRequired,
  active: PropTypes.oneOf(["left", "right"]).isRequired,
  onPress: PropTypes.func.isRequired,
}

TabButton.propTypes = {
  title: PropTypes.string.isRequired,
  active: PropTypes.oneOf(["left", "right"]).isRequired,
  onPress: PropTypes.func.isRequired,
  pos: PropTypes.oneOf(["left", "right"]).isRequired,
}

export default TabbedHeader
