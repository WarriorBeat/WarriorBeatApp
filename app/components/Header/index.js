/**
 * Header/index.js
 * Index Exports for Header Component
 * components
 */

import TabbedHeader from "./TabbedHeader"
import Header from "./Header"

export { Header, TabbedHeader }
