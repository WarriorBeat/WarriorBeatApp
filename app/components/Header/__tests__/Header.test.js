/**
 * __tests__/Header.test.js
 * Component Test
 * tests
 */

import React from "react"
import { shallow } from "enzyme"
import Header from "components/Header/Header"
import UIStore from "stores/uiStore"

const uiStore = new UIStore({})

describe("Testing Header Component", () => {
  it("renders as expected", () => {
    const wrapper = shallow(<Header.wrappedComponent uiStore={uiStore} />)
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ active: true })
    expect(wrapper).toMatchSnapshot()
  })
})
