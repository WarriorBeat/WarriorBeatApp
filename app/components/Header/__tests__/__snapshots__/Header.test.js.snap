// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component renders as expected 1`] = `
<Themed.Header
  barStyle="dark-content"
  leftComponent={
    <Return
      onPress={[Function]}
    />
  }
/>
`;

exports[`Testing Header Component renders as expected 2`] = `
<Themed.Header
  barStyle="light-content"
  leftComponent={
    <Return
      onPress={[Function]}
    />
  }
/>
`;
